### PR TITLE
[Localization] Partial Localization of Quick Claw Modifier

### PR DESCRIPTION
**Pull request by kvngrnr**
## What are the changes?
The message triggered by the Quick Claw modifier is hard coded right now. To address this, this PR starts by providing an english and german localization for the effect and uses the i18n plugin to show the correct translation to the user.
Additionally, the shop item for the quick claw got its correct german name (was just the english name before).

## Why am I doing these changes?
Having a hard coded english message might pose a problem to players with insufficient english skills as they might not understand the message. This could lead to frustration if, for example, the enemy pokemon uses their quick claw and the player does not understand why the were outsped.

## What did change?
The message text indicating that a pokemon just used its quick claw ("[PokemonName] used its Quick Claw to move faster!") was hard coded like this.
This PR starts by adding german and english localization for this text using the i18n plugin. Additionally, the german name for the quick claw item was added (was just the english name before).
**Unfortunately, I was not able to find any resources online on what this message is like in other languages in the main games, so I left out the localization for other languages for now.**

### Screenshots/Videos
Old version (german language but english message):
image

New version in german:
image

New english version in english:
image

New german name of item in shop:
image

**No I did not change the maximum held item count for the quick claw despite what the images suggest. That was just via the override file for simpler testing.**

## How to test the changes?
1. Add `{name: "QUICK_CLAW", count: 10}` to the `STARTING_HELD_ITEMS_OVERRIDE` and `"QUICK_CLAW"` to `ITEM_REWARD_OVERRIDE` in the `src/overrides.ts` file.
2. Start a new run.
3. Click an attack and check the displayed message.
4. Defeat a pokemon and check the first reward item's name.
5. Repeat in german and english.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by kvngrnr** [comment]
As stated in the "what did change" section, I was not able to find any translations of the quick claw trigger message online, so I left out the localization for other languages out for now. If anyone can provide me with valid translations, I'll be happy to add them!

File: src/locales/de/modifier-type.ts [hunk 1]
```diff
@@ -206,7 +206,7 @@ export const modifierType: ModifierTypeTranslationEntries = {
 
     "FOCUS_BAND": { name: "Fokusband", description: "Fügt eine 10% Chance hinzu, dass Angriffe die zur Kampfunfähigkeit führen mit 1 KP überlebt werden" },
 
-    "QUICK_CLAW": { name: "Quick Claw", description: "Fügt eine 10% Change hinzu als erster anzugreifen. (Nach Prioritätsangriffen)" },
+    "QUICK_CLAW": { name: "Flinkklaue", description: "Fügt eine 10% Change hinzu als erster anzugreifen. (Nach Prioritätsangriffen)" },
```
Comment (CodeTappert) [inline src/locales/de/modifier-type.ts:209]:
```suggestion
    "QUICK_CLAW": { name: "Flinkklaue", description: "Fügt eine 10% Change hinzu als Erster anzugreifen. (Nach Prioritätsangriffen)" },
```

**Comment by CodeTappert** [comment]
Also you should add the english text to all langauges. So it can later be translated easily

File: src/locales/de/modifier-type.ts (hunk 1 above)
Comment (kvngrnr) [reply src/locales/de/modifier-type.ts]: Good find, done

**Comment by kvngrnr** [comment]
@CodeTappert done, thanks for the input

File: src/locales/ko/battle.ts [hunk 2]
```diff
@@ -62,4 +62,5 @@ export const battle: SimpleTranslationEntries = {
   "drainMessage": "{{pokemonName}}[[로]]부터\n체력을 흡수했다!",
   "regainHealth": "{{pokemonName}}[[는]]\n기력을 회복했다!",
   "fainted": "{{pokemonNameWithAffix}}[[는]] 쓰러졌다!",
+  "useQuickClaw": "{{pokemonName}} used its Quick Claw to move faster!"
```
Comment (ddullgi) [inline src/locales/ko/battle.ts:65]:
```suggestion
  "useQuickClaw": "{{pokemonName}}[[은]] 선제공격손톱으로\n행동이 빨라졌다!"
```

**Review by ddullgi — CHANGES_REQUESTED** [review]

**Comment by EnochG1** [comment]
@kvngrnr 
Could you merge branch 'main' into your branch?
I want to continue this PR.
Or if it's difficult for you to continue this PR on your own, I would mirror yours and continue.

**Comment by kvngrnr** [comment]
@ EnochG1
Done. I actually forgot this PR even existed ^^

**Comment by EnochG1** [comment]
@kvngrnr
Thanks a lot!
I would share this PR to Translation team.
Also, if other language's translation are suggested, you can apply it just clicking this button.
image

File: src/locales/zh_TW/battle.ts [hunk 3]
```diff
@@ -130,4 +130,5 @@ export const battle: SimpleTranslationEntries = {
   "battlerTagsSaltCuredLapse": "{{pokemonNameWithAffix}} 受到了{{moveName}}的傷害！",
   "battlerTagsCursedOnAdd": "{{pokemonNameWithAffix}}削減了自己的體力，並詛咒了{{pokemonName}}！",
   "battlerTagsCursedLapse": "{{pokemonNameWithAffix}}正受到詛咒！",
+  "useQuickClaw": "{{pokemonName}} used its Quick Claw to move faster!"
```
Comment (40chyan) [inline src/locales/zh_TW/battle.ts:133]:
```suggestion
  "useQuickClaw": "{{pokemonName}}用了先製之爪後，\n行動變快了！"
```
zh_tw

File: src/locales/zh_CN/battle.ts [hunk 4]
```diff
@@ -132,5 +132,6 @@ export const battle: SimpleTranslationEntries = {
   "battlerTagsSaltCuredOnAdd": "{{pokemonNameWithAffix}}\n陷入了盐腌状态！",
   "battlerTagsSaltCuredLapse": "{{pokemonNameWithAffix}}\n受到了{{moveName}}的伤害！",
   "battlerTagsCursedOnAdd": "{{pokemonNameWithAffix}}削减了自己的体力，\n并诅咒了{{pokemonName}}！",
-  "battlerTagsCursedLapse": "{{pokemonNameWithAffix}}\n正受到诅咒！"
+  "battlerTagsCursedLapse": "{{pokemonNameWithAffix}}\n正受到诅咒！",
+  "useQuickClaw": "{{pokemonName}} used its Quick Claw to move faster!"
```
Comment (40chyan) [inline src/locales/zh_CN/battle.ts:136]:
```suggestion
  "useQuickClaw": "{{pokemonName}}用了先制之爪后，行动变快了！"
```
zh_cn

**Review by 40chyan — COMMENTED** [review]
Yep we translation team will submit translated texts to you, you can also find them on
[ABCBOY pokemon corpus](https://abcboy101.github.io/poke-corpus)
image

but in most cases we suggest you leave english for each language and ask us to submit. We have to check the grammar

File: src/locales/pt_BR/battle.ts [hunk 5]
```diff
@@ -132,5 +132,6 @@ export const battle: SimpleTranslationEntries = {
   "battlerTagsSaltCuredOnAdd": "{{pokemonNameWithAffix}} está sendo curado com sal!",
   "battlerTagsSaltCuredLapse": "{{pokemonNameWithAffix}} foi ferido pelo {{moveName}}!",
   "battlerTagsCursedOnAdd": "{{pokemonNameWithAffix}} cortou seus PS pela metade e amaldiçoou {{pokemonName}}!",
-  "battlerTagsCursedLapse": "{{pokemonNameWithAffix}} foi ferido pelo Curse!"
+  "battlerTagsCursedLapse": "{{pokemonNameWithAffix}} foi ferido pelo Curse!",
+  "useQuickClaw": "{{pokemonName}} used its Quick Claw to move faster!"
```
Comment (josericardo-fo) [inline src/locales/pt_BR/battle.ts:136]:
```suggestion
  "useQuickClaw": "{{pokemonName}} usou sua Garra Rápida para atacar primeiro!"
```

File: src/locales/en/battle.ts [hunk 6]
```diff
@@ -132,5 +132,6 @@ export const battle: SimpleTranslationEntries = {
   "battlerTagsSaltCuredOnAdd": "{{pokemonNameWithAffix}} is being salt cured!",
   "battlerTagsSaltCuredLapse": "{{pokemonNameWithAffix}} is hurt by {{moveName}}!",
   "battlerTagsCursedOnAdd": "{{pokemonNameWithAffix}} cut its own HP and put a curse on the {{pokemonName}}!",
-  "battlerTagsCursedLapse": "{{pokemonNameWithAffix}} is afflicted by the Curse!"
+  "battlerTagsCursedLapse": "{{pokemonNameWithAffix}} is afflicted by the Curse!",
+  "useQuickClaw": "{{pokemonName}} used its Quick Claw to move faster!"
```
Comment (Arxxer) [inline src/locales/en/battle.ts:136]:
```suggestion
  "useQuickClaw": "{{pokemonName}} can act faster than normal, thanks to its Quick Claw!"
```

**Review by Arxxer — COMMENTED** [review]
Hello! 

I'm suggesting a change for how the Quick Claw activation message is displayed to match how it actually is in all the original Pokemon games.

I did notice in Poke Corpus how in English it's displayed like [this ](https://abcboy101.github.io/poke-corpus/#query=can+act+faster+than+normal%2C+thanks+to&regex=false&caseInsensitive=true&common=true&script=true&collections=Emerald%2CDiamondPearl%2CPlatinum%2CHeartGoldSoulSilver%2CBlackWhite%2CBlack2White2%2CXY%2COmegaRubyAlphaSapphire%2CSunMoon%2CUltraSunUltraMoon%2CLetsGoPikachuLetsGoEevee%2CSwordShield%2CBrilliantDiamondShiningPearl%2CLegendsArceus%2CScarletViolet%2CColosseum%2CXD%2CBattleRevolution%2CBox%2CRanch%2CBank%2CHOME&languages=en-US%2Ces-ES%2Cen-GB) while trying to translate, and went ahead and also checked of others images in-game to check if it's actually accurate and it seems like so, so hopefully it isn't much trouble to do the change. 

Cheers!

File: src/locales/en/battle.ts (hunk 6 above)
Comment (kvngrnr) [reply src/locales/en/battle.ts:136]: Looks good to me.
@EnochG1 what do you and the translation team say? I could apply these translations from [this](https://abcboy101.github.io/poke-corpus/#query=can+act+faster+than+normal%2C+thanks+to&regex=false&caseInsensitive=true&common=true&script=true&collections=Emerald%2CDiamondPearl%2CPlatinum%2CHeartGoldSoulSilver%2CBlackWhite%2CBlack2White2%2CXY%2COmegaRubyAlphaSapphire%2CSunMoon%2CUltraSunUltraMoon%2CLetsGoPikachuLetsGoEevee%2CSwordShield%2CBrilliantDiamondShiningPearl%2CLegendsArceus%2CScarletViolet%2CColosseum%2CXD%2CBattleRevolution%2CBox%2CRanch%2CBank%2CHOME&languages=en-US%2Ces-ES%2Cen-GB) link and you guys could check if its fine?

Comment (EnochG1) [reply src/locales/en/battle.ts:136]: I agree with Arxxer's opinion. It's not MUST but if original game has messages, it's RECOMMENDED that you try to modify them as closely as possible.

**Comment by kvngrnr** [comment]
Alright @EnochG1, I updated the translations again. Please have the translation team carefully check all texts, I really just copied all from poke corpus. I can only validate english and german myself.

And thanks @Arxxer for your recommendation!

**Comment by CodeTappert** [comment]
German text is fine

**Comment by EnochG1** [comment]
@kvngrnr
Can you revert change of Korean?
Korean has special postposition logic(which is [[는]]) but you removed it and line break.
Before your modification(ddullgi) was right.

**Review by EnochG1 — APPROVED** [review]

File: src/locales/pt_BR/battle.ts [hunk 7]
```diff
@@ -132,5 +132,6 @@ export const battle: SimpleTranslationEntries = {
   "battlerTagsSaltCuredOnAdd": "{{pokemonNameWithAffix}} está sendo curado com sal!",
   "battlerTagsSaltCuredLapse": "{{pokemonNameWithAffix}} foi ferido pelo {{moveName}}!",
   "battlerTagsCursedOnAdd": "{{pokemonNameWithAffix}} cortou seus PS pela metade e amaldiçoou {{pokemonName}}!",
-  "battlerTagsCursedLapse": "{{pokemonNameWithAffix}} foi ferido pelo Curse!"
+  "battlerTagsCursedLapse": "{{pokemonNameWithAffix}} foi ferido pelo Curse!",
+  "useQuickClaw": "{{pokemonName}} usou sua {{quickClaw}} para atacar primeiro!"
```
Comment (josericardo-fo) [inline src/locales/pt_BR/battle.ts:136]:
```suggestion
  "useQuickClaw": "{{pokemonName}} se move mais rápido que o normal graças a sua {{quickClaw}}!"
```

**Comment by kvngrnr** [comment]
Pipeline failed for no reason. Is there a way to re-trigger the checks without a new commit?

**Comment by EnochG1** [comment]
@kvngrnr 
You should ask to dev admin if you want to do without a new commit.

File: src/modifier/modifier.ts [hunk 8]
```diff
@@ -9,7 +9,7 @@ import { addTextObject, TextStyle } from "../ui/text";
 import { Type } from "../data/type";
 import { EvolutionPhase } from "../evolution-phase";
 import { FusionSpeciesFormEvolution, pokemonEvolutions, pokemonPrevolutions } from "../data/pokemon-evolutions";
-import {getPokemonMessage, getPokemonNameWithAffix} from "../messages";
+import { getPokemonMessage, getPokemonNameWithAffix } from "../messages";
```
Comment (EnochG1) [inline src/modifier/modifier.ts:12]:
```suggestion
import { getPokemonNameWithAffix } from "../messages";
```

**Review by EnochG1 — COMMENTED** [review]
But I think we can fix it with this commit. ('main' is changed so getPokemonMessage don't need anymore)

**Comment by NicusPulcis** [comment]
@kvngrnr hey man i've fixed the italian translation in my recent PR on this project of yours (it was saying "quick claw moved faster thanks to pokemon name" lol)

**Comment by kvngrnr** [comment]
@NicusPulcis fixed, thank you

**Comment by NicusPulcis** [comment]
@kvngrnr aaalright

**Comment by EnochG1** [comment]
@kvngrnr
Could you merge branch 'beta' into your branch to resolve conflict?

**Comment by EnochG1** [comment]
@kvngrnr,
beta branch was reset after main release so old PRs needed to be reset.
Can you delete your local branch and get it from scratch?
If it's hard to do, I would mirror your task. Feel free and choose whichever you're more comfortable with.

**Comment by Tempo-anon** [comment]
Closing this PR for now due to all the merge conflicts. Please either fix those or make a new PR with the changes.

**Pull request closed without merging**